Binary files aa/drivers/media/platform/synopsys/.DS_Store and bb/drivers/media/platform/synopsys/.DS_Store differ
diff -ruN aa/drivers/media/platform/synopsys/hdmirx/snps_hdmirx.c bb/drivers/media/platform/synopsys/hdmirx/snps_hdmirx.c
--- aa/drivers/media/platform/synopsys/hdmirx/snps_hdmirx.c	2026-02-09 14:27:48
+++ bb/drivers/media/platform/synopsys/hdmirx/snps_hdmirx.c	2026-02-09 14:27:56
@@ -619,6 +619,8 @@
 
 static void hdmirx_plugout(struct snps_hdmirx_dev *hdmirx_dev)
 {
+	struct hdmirx_stream *stream = &hdmirx_dev->stream;
+
 	if (!hdmirx_dev->plugged)
 		return;
 
@@ -639,6 +641,17 @@
 			   PHY_PDDQ, HDMI_DISABLE);
 	hdmirx_writel(hdmirx_dev, PHYCREG_CONFIG0, 0x0);
 	cancel_delayed_work(&hdmirx_dev->delayed_work_res_change);
+
+	/*
+	 * If V4L2 streaming was active when we killed DMA, signal a
+	 * fatal error on the vb2 queue so that any blocked DQBUF
+	 * returns immediately with POLLERR/POLLHUP instead of
+	 * hanging on EAGAIN until userspace timeout.
+	 */
+	mutex_lock(&hdmirx_dev->stream_lock);
+	if (vb2_is_streaming(&stream->buf_queue))
+		vb2_queue_error(&stream->buf_queue);
+	mutex_unlock(&hdmirx_dev->stream_lock);
 
 	/* will be NULL on driver removal */
 	if (hdmirx_dev->rgb_range)
